### PR TITLE
dockerize

### PR DESCRIPTION
**Pull request by paulegradie**
### Description

This is ~VERY CLOSE~ to working correctly. 

Running this with `npm run dev-setup` will sort you right. ❤️ 

--- 
Old News
The changes to the docker-compose.yml will cause the nextJS app to be built inside of a docker container, which will watch the local project directory for changes and then update accordingly (so you can make changes without having to run `docker compose down; docker compose up` each time you make a change. Pretty nifty.

The app, database, and cli are networked, so they can speak to each other on the internal docker network.

To get it going, run

`docker compose up --build -d`

That will build a local image, and then start your services in detached mode.

To set up your database, do the usual thing of:

`npm run db:dev:refresh`

At this point, you'll noticd that we get an error 

`Error: P1001: Can't reach database server at `postgres-server`:`5432``

because the prisma.schema is trying to use the DATABASE_URL in the .env.development file that points to:

`postgres-server:5432` instead of localhost:5432

If you go into `.env.development` and change the connection string:

```
# DATABASE_URL=postgresql://postgres:Password01!@postgres-server:5432/dev
DATABASE_URL=postgresql://postgres:Password01!@localhost:5432/dev

// notice postgres-server:5432/dev VS localhost:5432/dev
```
and then do `docker compose down; docker compose up`, you'll notice that you can successfully now run 

`npm run db:dev:refresh`.

Once you've got that - you can run the app normally.

## TODO:
~How can we run docker compose up, and then run the setup script - without having to switch between local and docker network database conneciton string?~

~Solve that, and this will be fucking slick dev experience.~

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -8,25 +8,25 @@ datasource db {
...
 model Account {
-    id                 Int       @id @default(autoincrement())
-    user_name          String    @unique
-    email              String    @unique
-    password           String?
-    stripe_customer_id String    @unique
-    usage              Usage?
-    created_at         DateTime  @default(now())
-    updated_at         DateTime? @updatedAt
+    id               Int       @id @default(autoincrement())
```
Comment (paulegradie) [inline prisma/schema.prisma:11]:
```
datasource db {
    provider = "postgresql"
    url      = env("DATABASE_URL")
}
```
We might need a separate command for refreshing the database to point to a file that changes this database url to use localhost

**Comment by paulegradie** [comment]
One option appears to be to create a local docker network to use using

`docker network create my-network`

and then use that network in the app and outside the app. So we could just have a setup script that creates the network, and then just invokes the rest of the commands to get things stood up in a single command.

**Comment by Gwash3189** [comment]
@paulegradie would it be possible to add a specific docker-compose entry that runs the `db:dev:refresh` command? That way, we can ensure that the new entry is dependent on the postgres image starting up, and we can ensure that it can talk to it over the same docker-network?

**Comment by Gwash3189** [comment]
@paulegradie Or, we could change the command in the `package.json` to load multiple `.env` files. something like `dotenv -e .env.development -e .env.database.development -- ...`. Then add the correct DATABASE_URL to the `.env.database.development` file

**Comment by paulegradie** [comment]
I'm thinking along the lines of the second option. Here what I had:
```
        "db:dev:migrate": "dotenv -e ./scripts/.env.external -- npx prisma migrate dev",
        "db:dev:reset": "dotenv -e ./scripts/.env.external -- npx prisma migrate reset",
        "db:dev:refresh": "npm run db:dev:reset && npm run db:dev:migrate && npm run setup-db",
```

So i"ve added an .env.external to the scripts dir (since itsreally only specific to that setup bit, but can move it

**Comment by paulegradie** [comment]
@Gwash3189 have a look.

You should be able to just run

`npm run dev-setup`

**Review by Gwash3189 — APPROVED** [review]
Ooof - amazing work.

**Pull request merged**